packages/api/src/routes/local.ts

```diff
@@ -1,60 +1,55 @@
+ * This is a local development route only, and uses the same logic as the `cw-doc-contribution` lambda.
  *
  * Used to test the document retrieval part of the contribution logic for CommonWell.
  *
  * NOTE: For CW to send a request to your local env, you need to rig the flow in 2 places:
  *   1. In `cw-process-request`, the resulting bundle should have the localhost address be replaced with your ngrok address.
  *   2. In the CW portal, update your organization to use the <your-ngrok-address/oauth/fhir> under Gateway > FHIR R4 Endpoint.
  *
- * @param req.query.fileName The file name to get the signed URL for. The value for this comes from the
+ * @param req.query.fileName The file name to retrieve the document for. The value for this comes from the
  *  `oauth/fhir` route that responds to incoming document queries.
- * @returns A redirect to the signed URL for the file.
+ * @returns A FHIR Binary resource with the document data.
  */
 router.get(
   "/doc-contribution/commonwell",
   asyncHandler(async (req: Request, res: Response) => {
-    {
-      if (!Config.isDev()) {
-        throw new BadRequestError("This route is only available in dev environment");
-      }
-
-      try {
-        const s3Utils = new S3Utils(Config.getAWSRegion());
+    if (!Config.isDev()) {
+      throw new BadRequestError("This route is only available in dev environment");
+    }
 
-        const fileName = getFrom("query").orFail("fileName", req);
-        if (fileName.trim().length <= 0) {
-          return res.status(400).send("Missing fileName query parameter");
-        }
-        console.log(`File name: ${fileName}`);
+    const startedAt = Date.now();
+    try {
+      const fileName = getFrom("query").orFail("fileName", req);
+      const s3Utils = new S3Utils(Config.getAWSRegion());
+      const bucketName = Config.getMedicalDocumentsBucketName();
 
-        const key = fileName.startsWith("/") ? fileName.slice(1) : fileName;
-        if (!key || key.trim().length <= 0) {
-          return res.status(400).send("Invalid fileName query parameter");
-        }
+      const binary = await retrieveDocumentForCommonWell({
+        fileName,
+        s3Utils,
+        bucketName,
+      });
 
-        const url = await s3Utils.getSignedUrl({
-          bucketName: Config.getMedicalDocumentsBucketName(),
-          fileName: key,
-          durationSeconds: 60,
-        });
-        return res.status(301).redirect(url);
-      } catch (error) {
-        const msg = `Error processing DR from CW`;
-        console.log(`${msg}: ${errorToString(error)}`);
-        return res.status(500).send("Internal Server Error");
-      }
+      return res.status(200).type("application/octet-stream").send(binary);
+    } catch (error) {
+      const msg = `Error processing DR from CW`;
+      console.log(`${msg}: ${errorToString(error)}`);
+      return res.status(500).send("Internal Server Error");
```

Comment:
nit: why not just let the error bubble up to asynHandler?

Reply:
I'm sorry, but I realized why we have this: to truly replicate the behavior of the inbound flow, where we 500 on any internal errors, without exposing the details to the caller. Should we revert and leave it?